test/format/util.c

```diff
@@ -334,3 +345,87 @@ alter(void *arg)
     testutil_check(session->close(session, NULL));
     return (WT_THREAD_RET_VALUE);
 }
+
+/*
+ * lock_init --
+ *     Initialize abstract lock that can use either pthread of wt reader-writer locks.
+ */
+void
+lock_init(WT_SESSION *session, RWLOCK *lock)
+{
+    testutil_assert (lock->lock_type == LOCK_NONE);
+    testutil_assert (session != NULL);
+
+    if (g.c_wt_mutex) {
+	testutil_check(__wt_rwlock_init((WT_SESSION_IMPL *)session, &lock->l.wt));
+	lock->lock_type = LOCK_WT;
+    } else {
+	testutil_check(pthread_rwlock_init(&lock->l.pthread, NULL));
+	lock->lock_type = LOCK_PTHREAD;
+    }
+}
+
+/*
+ * lock_destroy --
+ *     Destroy abstract lock.
+ */
+void
+lock_destroy(WT_SESSION *session, RWLOCK *lock)
+{
+    if (g.c_wt_mutex) {
+	testutil_assert(lock->lock_type == LOCK_WT);
+	__wt_rwlock_destroy((WT_SESSION_IMPL *)session, &lock->l.wt);
+    } else {
+	testutil_assert(lock->lock_type == LOCK_PTHREAD);
+	testutil_check(pthread_rwlock_destroy(&lock->l.pthread));
+    }
+    lock->lock_type = LOCK_NONE;
+}
+
+/*
+ * lock_try_writelock
+ *     Try to get exclusive lock.  Fail immediately if not available.
+ */
+int
+lock_try_writelock(WT_SESSION *session, RWLOCK *lock)
+{
+    if (g.c_wt_mutex) {
+	testutil_assert(lock->lock_type == LOCK_WT);
+	return(__wt_try_writelock((WT_SESSION_IMPL *)session, &lock->l.wt));
+    } else {
+	testutil_assert(lock->lock_type == LOCK_PTHREAD);
+	return(pthread_rwlock_trywrlock(&lock->l.pthread));
+    }
+}
+
+/*
+ * lock_writelock --
+ *     Wait to get exclusive lock.
+ */
+void
+lock_writelock(WT_SESSION *session, RWLOCK *lock)
+{
+    if (g.c_wt_mutex) {
+	testutil_assert(lock->lock_type == LOCK_WT);
+	__wt_writelock((WT_SESSION_IMPL *)session, &lock->l.wt);
+    } else {
+	testutil_assert(lock->lock_type == LOCK_PTHREAD);
+	testutil_check(pthread_rwlock_wrlock(&lock->l.pthread));
+    }
+}
+
+/*
+ * lock_writeunlock --
+ *     Release an exclusive lock.
+ */
+void
+lock_writeunlock(WT_SESSION *session, RWLOCK *lock)
+{
+    if (g.c_wt_mutex) {
+	testutil_assert(lock->lock_type == LOCK_WT);
+	__wt_writeunlock((WT_SESSION_IMPL *)session, &lock->l.wt);
+    } else {
+	testutil_assert(lock->lock_type == LOCK_PTHREAD);
+	testutil_check(pthread_rwlock_unlock(&lock->l.pthread));
+    }
```

Comment:
I'd inline the actual lock/unlock, and not pay the function call.

Reply:
Done